assets/src/edit-story/index.js

```diff
@@ -41,7 +42,9 @@ const initialize = (id, config, flags) => {
   Modal.setAppElement(appElement);
   render(
     <FlagsProvider features={flags}>
-      <App config={config} />
+      <ErrorBoundary>
```

Comment:
This one we can't Undo. Are you ok relying on the default context of `useHistory` where `canUndo` will simply be `undefined`?

Reply:
Also it means losing all state from all providers, that have to re-init completely - e.g. reload the story from API, create new clean history, etc.

Reply:
At that level I don't show undo (it will not work), just the reload option is displayed.

Either way it was just an example that while we can do something like this, it will not work for all cases, and in most cases it will prevent you from using the editor in full capacity, undo will not fix that.

The real question is: what to do in case of tree error, asked [here](https://xwp.slack.com/archives/C0135V8LYJE/p1591345006182700). @dvoytenko @barklund WDYT?